src/common/mock-service.c

```diff
@@ -454,6 +456,68 @@ on_tell_me_your_name (TestFrobber *frobber,
 
 }
 
+static gboolean
+test_fd_pipe_readable (gint fd,
+                       GIOCondition condition,
+                       gpointer user_data)
+{
+  const char *expected = "Hello, fd";
+  char buffer[100];
+
+  g_assert (condition == G_IO_IN);
+
+  read (fd, buffer, 100);
```

Comment:
Thanks for the catch. I've added a similar check like I have for `write`. I think we can safely assume that we always get all data at once since this is a pipe and very little data.